packages/fuels-core/src/traits/signer.rs

```diff
@@ -0,0 +1,14 @@
+use async_trait::async_trait;
+use fuel_crypto::{Message, Signature};
+
+use crate::types::{bech32::Bech32Address, errors::Result};
+
+/// Trait for signing transactions and messages
+///
+/// Implement this trait to support different signing modes, e.g. hardware wallet, hosted etc.
+#[cfg_attr(target_arch = "wasm32", async_trait(?Send))]
+#[cfg_attr(not(target_arch = "wasm32"), async_trait)]
+pub trait Signer: std::fmt::Debug + Send + Sync + 'static {
```

Comment:
I'm not sure about the `Debug` bound.

 It is needed because our transaction builders are debug. 

Maybe we should do a custom impl of `Debug` for the TBs and remove this bound here? 

In our custom impl on TBs we can just say `[Signer(PUB_ADDR), Signer(PUB_ADDR), ...]`.

Reply:
Added custom `Debug` for `dyn Signer`.